docs/rules/sort-imports.md

```diff
@@ -0,0 +1,167 @@
+
+## Rule Details
+
+This rule checks all import declarations and verifies that all imports using the same member syntax are sorted alphabetically by the first member.
+
+This rule also checks the ordering of the import declarations using the same member syntax style. See option ```groupByMemberSyntax``` for more details.
+
+The default configuration of the rule
+
+- is case-sensitive.
+- checks member ordering.
+- checks ordering of import declaration by used member syntax: ```"entire", "all", "multiple", "single"```
+
+The following patterns are considered problems:
+
+```js
+
+/*eslint sort-imports: 2*/
+import b from 'foo.js';
+import a from 'bar.js'; /*error Imports should be sorted alphabetically.*/
+
+
+/*eslint sort-imports: 2*/
+import a from 'foo.js';
+import A from 'bar.js'; /*error Imports should be sorted alphabetically.*/
+
+/*eslint sort-imports: 2*/
+import {b, c} from 'foo.js';
+import {a, b} from 'bar.js'; /*error Imports should be sorted alphabetically.*/
+
+/*eslint sort-imports: 2*/
+import a from 'foo.js';
+import {b, c} from 'bar.js'; /*error Imports should be grouped by the member syntax. Use 'multiple' before 'single' member syntax.*/
+
+/*eslint sort-imports: 2*/
+import a from 'foo.js';
+import * as b from 'bar.js'; /*error Imports should be grouped by the member syntax. Use 'all' before 'single' member syntax. */
+
+/*eslint sort-imports: 2*/
+import {b, a, c} from 'foo.js' /*error Members of an import declaration should be sorted alphabetically.*/
+
+```
+
+The following patterns are not considered problems:
+
+```js
+
+/*eslint sort-imports: 2*/
+import a from 'foo.js';
+import b from 'bar.js';
+import c from 'baz.js';
+
+/*eslint sort-imports: 2*/
+import 'foo.js'
+import * from 'bar.js';
+import {a, b} from 'baz.js';
+import c from 'qux.js';
+
+/*eslint sort-imports: 2*/
+import {a, b, c} from 'foo.js'
+
+```
+
+
+## Rule Options
+
+```
+"sort-imports": [<enabled>, {
+    "ignoreCase": <boolean>,
+    "groupByMemberSyntax": <array<enum>>,
+    "ignoreMemberSort": <boolean>
+}]
+```
+
+### `ignoreCase`
+
+When `true` the rule ignores the case-sensitivity of the imports local name.
+
+The following patterns are considered problems:
+
+```js
+/*eslint sort-imports: [2, { "ignoreCase": true }]*/
+
+import B from 'foo.js';
+import a from 'bar.js';
+```
+
+
+The following patterns are not considered problems:
+
+```js
+/*eslint sort-imports: [2, { "ignoreCase": true }]*/
+
+import a from 'foo.js';
+import B from 'bar.js';
+import c from 'baz.js';
+```
+
+### `groupByMemberSyntax`
+
+The import statement supports multiple member syntax options. For more details see [MDN - import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)
+
+This rule checks the ordering of the import declarations using the same member syntax style. The default ordering of the member syntax is ```["entire", "all", "multiple", "single"]```. The ordering can be configured, the list shall always contain 4 items.
```

Comment:
Maybe "entire" is not so bad at all.

### entire
Imports entire module, no member bindings.
``` javascript
import "my-module";
```

### all
Imports all members from module.
```javascript
import * as root from "my-module.js";
```

### multiple
Imports multiple members from a module.
```javascript
import {foo, bar} from "my-module.js";
```

### single
Imports one member from a module.
```javascript
import foo from "my-module.js";
```

Or rename all to "entireModule", "allMembers", "multipleMembers", "singleMembers"?